hashing/te-tag-query-java/com/facebook/threatexchange/TETagQuery.java

```diff
@@ -904,78 +904,248 @@ public void handle(
-      o.printf("-I                     Take indicator text from standard input, one per line.\n");
-      o.printf("Exactly one of -i or -I is required.\n");
-      o.printf("-t|--type {...}\n");
-      o.printf("-d|--description {...}\n");
-      o.printf("-l|--share-level {...}\n");
-      o.printf("-p|--privacy-type {...}\n");
-      o.printf("-y|--severity {...}\n");
-      o.printf("\n");
-      o.printf("Optional:\n");
-      o.printf("-h|--help\n");
-      o.printf("--dry-run\n");
-      o.printf("-m|--privacy-members {...} If privacy-type is HAS_WHITELIST these must be\n");
-      o.printf("                       comma-delimited app IDs. If privacy-type is\n");
-      o.printf("                       HAS_PRIVACY_GROUP these must be comma-delimited\n");
-      o.printf("                       privacy-group IDs.\n");
-      o.printf("--tags {...}           Comma-delimited. Overwrites on repost.\n");
-      o.printf("--related-ids-for-upload {...} Comma-delimited. IDs of descriptors (which must\n");
-      o.printf("                       already exist) to relate the new descriptor to.\n");
-      o.printf("--related-triples-json-for-upload {...} Alternate to --related-ids-for-upload.\n");
-      o.printf("                       Here you can uniquely the relate-to descriptors by their\n");
-      o.printf("                       owner ID / indicator-type / indicator-text, rather than\n");
-      o.printf("                       by their IDs. See README.md for an example.\n");
-      o.printf("--reactions-to-add {...} Example for add/remove: INGESTED,IN_REVIEW\n");
-      o.printf("--reactions-to-remove {...}\n");
-      o.printf("-c|--confidence {...}\n");
-      o.printf("-s|--status {...}\n");
-      o.printf("-r|--review-status {...}\n");
-      o.printf("--first-active {...}\n");
-      o.printf("--last-active {...}\n");
-      o.printf("--expired-on {...}\n");
-      o.printf("\n");
-      o.printf("Please see the following for allowed values in all enumerated types except reactions:\n");
-      o.printf("https://developers.facebook.com/docs/threat-exchange/reference/submitting\n");
-      o.printf("Please see the following for allowed values in all enumerated types in reactions:\n");
-      o.printf("https://developers.facebook.com/docs/threat-exchange/reference/reacting\n");
-      System.exit(exitCode);
+      this.commonPosterUsage(exitCode, POSTER_NAME_SUBMIT);
     }
 
     @Override
     public void handle(
-      String[] args,
+      String[] argsAsArray,
       int numIDsPerQuery,
       boolean verbose,
       boolean showURLs,
       DescriptorFormatter descriptorFormatter
     ) {
       boolean dryRun = false;
       boolean indicatorTextFromStdin = false;
-      DescriptorPostParameters  params = new DescriptorPostParameters();
+      DescriptorPostParameters postParams = new DescriptorPostParameters();
+      ArrayList<String> args = new ArrayList<String>(Arrays.asList(argsAsArray));
```

Comment:
Non-blocking

Would using an Iterator in place of a list here be helpful? I'm thinking it could help clean up some of the code that consumes additional arguments while parsing through the command line options be using `hasNext()` and `next()` instead of modifying the array list.

Reply:
hmm, food for thought. seems kind of six of one, half a dozen of the other ...